nltk/twitter/twitterclient.py

```diff
@@ -0,0 +1,457 @@
+                continue
+
+    def filter(self, track='', follow='', lang='en', locations=None):
+        """
+        Wrapper for 'statuses / filter' API call
+        """
+        while self.do_continue:
+            '''
+            Stream in an endless loop until limit is reached
+            see twython issue 288
+                https://github.com/ryanmcgrath/twython/issues/288
+                colditzjb commented on 9 Dec 2014
+            '''
+            try:
+                if track == '' and follow == '' and locations == None:
+                    msg = "Please supply a value for 'track', 'follow' or 'locations'."
+                    raise ValueError(msg)
+                self.statuses.filter(track=track, follow=follow, lang=lang,
+                                     locations=locations)
+            except requests.exceptions.ChunkedEncodingError as e:
+                if e is not None:
+                    print("Error (stream will continue): {0}".format(e))
+                continue
+
+
+
+
+class Query(Twython):
+    """
+    Retrieve data from the Twitter REST API.
+    """
+    def __init__(self, app_key, app_secret, oauth_token,
+                 oauth_token_secret):
+        self.handler = None
+        self.do_continue = True
+        Twython.__init__(self, app_key, app_secret, oauth_token, oauth_token_secret)
+
+    def register(self, handler):
+        """
+        Register a method for handling tweets.
+
+        :param TweetHandlerI handler: method for viewing or writing tweets to a file.
+        """
+        self.handler = handler
+
+    def _lookup(self, infile, verbose=True):
+        """
+        :param infile: name of a file consisting of Tweet IDs, one to a line
+        :return: iterable of Tweet objects
+        """
+        with open(infile) as f:
+            ids = [line.rstrip() for line in f]
+        if verbose:
+            print("Counted {0} Tweet IDs in {1}.".format(len(ids), infile))
+
+        id_chunks = [ids[i:i+100] for i in range(0, len(ids), 100)]
+        """
+        The Twitter endpoint takes lists of up to 100 ids, so we chunk
+        the ids.
+        """
+
+        listoflists = [self.post('statuses/lookup', {'id': chunk}) for chunk
+                       in id_chunks]
+        return itertools.chain.from_iterable(listoflists)
+
+
+    def lookup(self, infile, outfile, verbose=True):
+        """
+        Given a file containing a list of tweetIDs, fetch the corresponding
+        Tweets (if they haven't been deleted) and dump them in a file.
+
+        :param str infile: Name of a file consisting of tweetIDs, one to a line
+        :param str outfile: Name of file where JSON serialisations of fully\
+        expanded Tweets will be written.
+        """
+        tweets = self._lookup(infile, verbose=verbose)
+        count = 0
+
+        if os.path.isfile(outfile):
+            os.remove(outfile)
+
+        with open(outfile, 'a') as f:
```

Comment:
In the context of something as voluminous at Twitter data, I am rather wary of methods that only accept *paths* to files, rather than file-like objects that might incorporate compression. In general, I think it is bad Python design to admit paths exclusively to an I/O interface.